server/new-commiter.go

```diff
@@ -0,0 +1,163 @@
+	author string
+	date   time.Time
+	commit string
+}
+
+func (p *Plugin) executeNewCommiterCommand(commandArgs []string, args *model.CommandArgs) *model.AppError {
+	if len(commandArgs) != 2 {
+		return &model.AppError{
+			Id:         "Need two arguments",
+			StatusCode: http.StatusBadRequest,
+			Where:      "p.ExecuteCommand",
+		}
+	}
+
+	organisation := commandArgs[0]
+
+	since, err := time.Parse(shortFormWithDay, commandArgs[1])
+	if err != nil {
+		return &model.AppError{
+			Id:         fmt.Sprintf("Failed to parse since time: %v", err.Error()),
+			StatusCode: http.StatusBadRequest,
+			Where:      "p.ExecuteCommand",
+		}
+	}
+
+	org, _, err := p.client.Organizations.Get(context.Background(), organisation)
+	if err != nil {
+		return &model.AppError{
+			Id:         "Failed to fetch data",
+			StatusCode: http.StatusBadRequest,
+			Where:      "p.ExecuteCommand",
+		}
+	}
+
+	attachments := []*model.SlackAttachment{{
+		Title:      "Fetching new commiters since " + since.Format(shortFormWithDay),
+		Text:       waitText,
+		AuthorName: organisation,
+		AuthorIcon: org.GetAvatarURL(),
+		AuthorLink: fmt.Sprintf("https://github.com/%v", organisation),
+	}}
+
+	loadingPost := &model.Post{
+		ChannelId: args.ChannelId,
+		UserId:    p.botUserID,
+	}
+	model.ParseSlackAttachment(loadingPost, attachments)
+
+	loadingPost, appErr := p.API.CreatePost(loadingPost)
+	if appErr != nil {
+		return appErr
+	}
+
+	go p.updateNewCommitersPost(loadingPost, args.UserId, organisation, since)
+
+	return nil
+}
+
+func (p *Plugin) updateNewCommitersPost(post *model.Post, userID, org string, since time.Time) {
+	contributors, err := p.fetchContributorsFromOrg(org)
+	if err != nil {
+		p.API.LogError("failed to fetch data", "err", err.Error())
+
+		var message string
+		if _, ok := err.(*github.RateLimitError); ok {
+			message = "Hit rate limit. Please try again later."
+		} else {
+			message = "Failed to fetch data:" + err.Error()
+		}
+		post.Props["attachments"].([]*model.SlackAttachment)[0].Text = message
+	} else {
+		firstContributions := p.findFirstContributions(contributors)
+		p.filterContributionsByWeek(firstContributions, since)
+
+		var result []*firstContributionInfo
+		for commiter, contribution := range firstContributions {
+			commits, _ := p.fetchCommitsFromRepoByAuthorAndWeek(org, contribution.repo, commiter, contribution.contributionWeek)
+			firstCommit := commits[0]
+			if firstCommit.GetCommit().Committer.GetDate().After(since) {
+				result = append(result, &firstContributionInfo{firstCommit.GetAuthor().GetLogin(), firstCommit.GetCommit().Committer.GetDate(), firstCommit.GetHTMLURL()})
+			}
+		}
+
+		sort.Slice(result, func(i, j int) bool {
+			return result[i].date.Before(result[j].date)
+		})
+
+		var resultText string
+		for _, e := range result {
+			resultText += fmt.Sprintf("- [%[1]s](https://github.com/%[1]v)  [first commit](%s)  %s\n", e.author, e.commit, e.date.Format(shortFormWithDay))
```

Comment:
It would be helpful if the command would also return the repo e.g. `jseiser: first commit at 2020-06-02 on mattermost-helm`